main.go

```diff
@@ -1851,6 +1878,72 @@ func (git *repoSync) CallAskPassURL(ctx context.Context) error {
 	return nil
 }
 
+func (git *repoSync) RefreshAppToken(ctx context.Context, githubBaseURL, privateKey string, appID, installationID int) error {
```

Comment:
I'd like the entire set of functions to reflect "GitHubApp" rather than just app, so it's easier to read.  Could also do with a nice comment, please